### ISSUE TITLE
Add possibility to filter guest users in the user manager

### ISSUE THREAD

**Issue by IljaN** [owncloud/guests#16]

**Comment by IljaN** [comment]
@felixheidecke We need some UI in the group manager under "Everyone" to exclude certain groups (guests).
![usermanager](https://cloud.githubusercontent.com/assets/246983/23762193/9552204c-04f5-11e7-9508-ed50b042dd6c.png)

Maybe a input with a semicolon separated list.

The User Manager does following request to get the list for gorup "everyone":
```/index.php/settings/users/users?offset=0&limit=50&gid=&pattern=```

I added a additional parameter "excludeGroups" on the backend. So now you can do.
```/index.php/settings/users/users?offset=0&limit=50&gid=&pattern=&excludeGroups=guests```
or to exclude multiple groups
```/index.php/settings/users/users?offset=0&limit=50&gid=&pattern=&excludeGroups=guests;othergroup```

The changes have been pushed to the 1_move_jailing_code_to_core branch, I propose that we do all our core work in there.

**Comment by IljaN** [comment]
https://github.com/owncloud/core/commit/6c17519485a258a1fa464b08aabf06f7cbe06c5e

**Comment by PVince81** [comment]
> Maybe a input with a semicolon separated list.

No! This is too technical. Or did you mean have an autocomplete (select2?) field like in the admin page for sharing groups ?

Since it is currently only possible to filter by a single group, I don't think we should allow excluding more than a single group ?

So the only possible filters are:
- no filter
- filter by a single group (only show users in this group), this is when clickiing on a group
- filter a single group out (show all users except those in that group), UI to be designed

Thoughts ?

**Comment by felixheidecke** [comment]
> Since it is currently only possible to filter by a single group …

The top-right search field only filters by username … or am I doing this wrong?

**Comment by felixheidecke** [comment]
We could use the filter in the rop right to exclude groups. I'm thinking about doing it that way:
`!guests feli !felidae` This would exclude the guests, find felix but exclude felidae as well.
What do you think?

We could prepopulate the field with !guests

**Comment by IljaN** [comment]
@felixheidecke I think it still would be too technical, maybe add a input to the options in the lower left at the "Zahnrad" icon

@PVince81 what do you think?

**Comment by felixheidecke** [comment]
Zahnrad is too far away. Nobody would expect it there. If we want to implement a filter, it shouldn't be half-baked. Filtering groups 1 by 1 isn't enough I guess.

It's in the discretion of the user how technical he wants to get. E.g. google search. You could either just type away and look for _"funny kittens"_ or go all technical on it _"funny kittens OR cats @twitter -dogs"_

**Comment by felixheidecke** [comment]
It could be a simple AND search where "-" excludes. Don't you think?

**Comment by PVince81** [comment]
No, I don't think we should introduce extended filtering at this stage with expressions etc...

With "single group filter" I mean "Pick a group on the left sidebar and click on it". It will filter by this one group.

Then maybe have a way to swap it around and have that one group excluded.

**Comment by PVince81** [comment]
The idea of the cog icon (Zahnrad) is good too, but the use case is then mostly "put a group there that you never ever want to see in the list by default". So it's something to set only once. We'd set this to the guest group by default. If an admin does want to manage guests, they can go there and remove this default filter.

**Comment by PVince81** [comment]
Anything left to do here or does the latest state match the requirements now ?

**Comment by felixheidecke** [comment]
Currently … I don't see that requirement anywhere. No sure of this is up to date: https://github.com/owncloud/platform/issues/11

**Comment by PVince81** [comment]
moving to backlog then
